adapters/dmx/dmx.go

```diff
@@ -100,6 +102,12 @@ func (adapter *DmxAdapter) MakeRequests(request *openrtb.BidRequest, req *adapte
 		if dmxReq.App.ID != "" {
 			anyHasId = true
 		}
+		if anyHasId == false {
+			if idfa, valid := getIdfa(request); valid {
```

Comment:
Looks good. Please add a json test for the `idfa, valid := getIdfa(request); valid` condition.